### PR TITLE
fix(Header-v1): Added disabled state for Header breadcrumb items

### PR DESCRIPTION
**Pull request by adamhock**
- Addresses [this comment](https://github.com/iTwin/iTwinUI-react/pull/755#discussion_r950098171) made in #755
- Added `aria-disabled` to `iui-header-breadcrumb-item` to apply the proper styling.

### CODE REVIEW

File: packages/iTwinUI-react/src/core/Header/HeaderButton.tsx [hunk 1]
```diff
@@ -113,6 +115,7 @@ export const HeaderButton: HeaderButtonComponent = React.forwardRef(
       <li
         className='iui-header-breadcrumb-item'
         aria-current={isActive ? 'location' : undefined}
+        aria-disabled={disabled ? 'true' : undefined}
```
Comment (veekeys) [inline packages/iTwinUI-react/src/core/Header/HeaderButton.tsx:118]: Can you add disabled button to at least one of our stories? Would be good to have test for it :)

Comment (veekeys) [reply packages/iTwinUI-react/src/core/Header/HeaderButton.tsx:118]: And verify that all others work (basic, split and dropdown)

Comment (adamhock) [reply packages/iTwinUI-react/src/core/Header/HeaderButton.tsx:118]: I disabled the split button in the Full Header story. These are what each of the following disabled buttons look like (split, dropdown, and basic)

image

**Review by veekeys — APPROVED** [review]

**Review by r100-stack — APPROVED** [review]
Looks good! Only one thing I wondered about was that there are no unit tests. But maybe they are not needed for this change so it's okay

**Pull request merged**